Src/Fido2/AttestationFormat/Apple.cs

```diff
@@ -0,0 +1,148 @@
+                throw new Fido2VerificationException("Attestation format apple must have attestation statement");
+
+            if (null == Alg || true != Alg.IsNumber)
+                throw new Fido2VerificationException("Invalid apple attestation algorithm");
+
+            // Verify that x5c is present
+            if (null == X5c || CBORType.Array != X5c.Type || 0 == X5c.Count || null != EcdaaKeyId)
+                throw new Fido2VerificationException("Malformed x5c array in apple attestation statement");
+
+            // Validate certificate lifetimes
+            var enumerator = X5c.Values.GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                if (null == enumerator || null == enumerator.Current
+                    || CBORType.ByteString != enumerator.Current.Type
+                    || 0 == enumerator.Current.GetByteString().Length)
+                    throw new Fido2VerificationException("Malformed x5c cert found in apple attestation statement");
+
+                var x5ccert = new X509Certificate2(enumerator.Current.GetByteString());
+
+                // X509Certificate2.NotBefore/.NotAfter return LOCAL DateTimes, so
+                // it's correct to compare using DateTime.Now.
+                if (DateTime.Now < x5ccert.NotBefore || DateTime.Now > x5ccert.NotAfter)
+                    throw new Fido2VerificationException("Apple signing certificate expired or not yet valid");
+            }
+
+            // The attestation certificate attestnCert MUST be the first element in the array.
+            var attestnCert = new X509Certificate2(X5c.Values.First().GetByteString());
+
+            // Version MUST be set to 3
+            if (3 != attestnCert.Version)
+                throw new Fido2VerificationException("Apple x5c attestation certificate not V3");
+
+            // The Basic Constraints extension MUST have the CA component set to false
+            if (IsAttnCertCACert(attestnCert.Extensions))
+                throw new Fido2VerificationException("Attestation certificate has CA cert flag present");
+
+            // Verify that the x5c array contains the intermediate and leaf certificates for atttestation,
+            // starting from the credential certificate stored in the first data buffer in the array.
+            // Verify the validity of the certificates using Apple's Webauthn Root CA
+            var trustPath = X5c.Values
+                .Select(x => new X509Certificate2(x.GetByteString()))
+                .ToArray();
+
+            // If the authenticator is listed as in the metadata as one that should produce a basic full attestation, build and verify the chain
+            var entry = _metadataService?.GetEntry(AuthData.AttestedCredentialData.AaGuid);
+
+            // while conformance testing, we must reject any authenticator that we cannot get metadata for
+            if (_metadataService?.ConformanceTesting() == true && null == entry)

```

Comment:
Drop `== true`.

Reply:
And similar code a lot of other places, unless this is the upstream standard.

Reply:
This compares a _nullable_ boolean.